samples/FtpServer_Files/component.mk

```diff
@@ -0,0 +1,2 @@
+HWCONFIG := spiffs
+DISABLE_SPIFFS := 1
```

Comment:
Aren't these two lines conflicting? Isn't the first line saying - use spiffs, here is the hardware configuration and the second one saying, well actually tell the app that we are not using spiffs? 

Can you remove the second line and leave the first one?

Reply:
I'd like to get rid of [DISABLE_SPIFFS](http://localhost:8000/_inc/Sming/Components/spiffs/index.html?highlight=disable_spiffs#envvar-DISABLE_SPIFFS) entirely as it's misleading - it means "don't generate a filesystem image". Try running `make DISABLE_SPIFFS=0` and see what happens... (answer: "Unable to open directory 'files'").

This behaviour can be overridden by customising the 'spiffs0' partition so could be removed in future...

Reply:
Perhaps a better solution to this is to get rid of DISABLE_SPIFFS and replace it with `SPIFF_FILES=` instead. This has the same effect. (Try `make DISABLE_SPIFFS=0 SPIFF_FILES=`)

Reply:
Correction. DISABLE_SPIFFS and SPIFF_FILES have subtly different behaviour.

`SPIFF_FILES=` Create empty SPIFFS image (`make flash` will write this)
`DISABLE_SPIFFS = 1` Don't create a filesystem image (`make flash` will not write anything to the partition)

Reply:
> I'd like to get rid of DISABLE_SPIFFS entirely

Agreed. Add also error message error when DISABLE_SPIFFS=1 is used. Something like:
`$(error  "DISABLE_SPIFFS=1 is no longer allowed. You can fix the issue by deleting this option from your component.mk file or config.")`